dbms/src/Debug/MockStorage.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+#include <Core/ColumnsWithTypeAndName.h>
+#include <Storages/Transaction/TiDB.h>
+#include <common/types.h>
+
+#include <unordered_map>
+namespace DB::tests
+{
+using MockColumnInfo = std::pair<String, TiDB::TP>;
+using MockColumnInfoVec = std::vector<MockColumnInfo>;
+
+class MockTableIdGenerator : public ext::Singleton<MockTableIdGenerator>
+{
+public:
+    static MockTableIdGenerator & getInstance()
+    {
+        static MockTableIdGenerator table_id_generator;
+        return table_id_generator;
+    }
+    Int64 nextTableId()
```

Comment:
Can we ensure that this function can be protected by a lock? I think this function may be called by multi-threads.